src/screens/choice-exercises/WordChoiceExerciseScreen.tsx

```diff
@@ -55,11 +55,14 @@ const WordChoiceExerciseScreen = ({ navigation, route }: WordChoiceExerciseScree
   }
 
   return (
-    <SingleChoiceExercise
-      documents={documents}
-      documentToAnswers={documentToAnswers}
-      onExerciseFinished={onExerciseFinished}
-    />
+    documents !== null &&
+    documents.length !== 0 && (
```

Comment:
Why is this needed?

Reply:
it is not i guess. removed it.

Reply:
ok, it is needed, added it again. If loading is not finished, it needs to block rendering.

Reply:
But shouldn't there be a `loading` prop which does exactly that?

Reply:
Yeah it should and it is, but if removing i get ` Type 'DocumentType[] | null' is not assignable to type 'DocumentType[]'.   Type 'null' is not assignable to type 'DocumentType[]'.`

Because ReturnType<DocumentType[]> also allows `null`. I don't know if there is a good way to tell typescript that it will only be null if loading is true.

Reply:
oke sorry, i was talking about the first line, you are right, the second line can be replaced with `!loading`